formatter/formatSpecification.go

```diff
@@ -29,27 +29,47 @@ type formatter struct {
 	buffer bytes.Buffer
 }
 
+var nextItem gauge.Item
+
+func (formatter *formatter) SetNext(item gauge.Item) {
+	nextItem = item
+}
+
+func (formatter *formatter) GetNext() (item gauge.Item) {
+	return nextItem
+}
+
 func (formatter *formatter) Specification(specification *gauge.Specification) {
 }
 
-func (formatter *formatter) SpecHeading(specHeading *gauge.Heading) {
-	formatter.buffer.WriteString(FormatHeading(specHeading.Value, "="))
+func (formatter *formatter) Heading(heading *gauge.Heading) {
+	if heading.HeadingType == gauge.SpecHeading {
+		formatter.buffer.WriteString(FormatHeading(heading.Value, "="))
+	} else if heading.HeadingType == gauge.ScenarioHeading {
+		formatter.buffer.WriteString(FormatHeading(heading.Value, "-"))
+	}
 }
 
-func (formatter *formatter) SpecTags(tags *gauge.Tags) {
+func (formatter *formatter) Tags(tags *gauge.Tags) {
+	if !strings.HasSuffix(formatter.buffer.String(), "\n\n") {
+		formatter.buffer.WriteString("\n")
+	}
 	formatter.buffer.WriteString(FormatTags(tags))
+	if formatter.GetNext().Kind() != gauge.CommentKind || strings.TrimSpace(nextItem.(*gauge.Comment).Value) != "" {
+		formatter.buffer.WriteString("\n")
+	}
 }
 
-func (formatter *formatter) DataTable(table *gauge.Table) {
+func (formatter *formatter) Table(table *gauge.Table) {
 	formatter.buffer.WriteString(strings.TrimPrefix(FormatTable(table), "\n"))
 }
 
-func (formatter *formatter) ExternalDataTable(extDataTable *gauge.DataTable) {
-	formatter.buffer.WriteString(FormatExternalDataTable(extDataTable))
-}
-
-func (formatter *formatter) ContextStep(step *gauge.Step) {
-	formatter.Step(step)
+func (formatter *formatter) DataTable(dataTable *gauge.DataTable) {
+	if !dataTable.IsExternal {
+		formatter.Table(&(dataTable.Table))
+	} else {
+		formatter.buffer.WriteString(FormatExternalDataTable(dataTable))
```

Comment:
If `FormatExternalDataTable` is not used outside the package, it should not be exported.